### ISSUE TITLE
Evaluating on training data with the Object Detection API

### ISSUE THREAD

**Issue by iliasmansouri** [tensorflow/models#6069]
### System information
- **What is the top-level directory of the model you are using**:
research/object_detection/
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
yes as in modifying the pipeline.config for my needs
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Mac OS with Cloud ML
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
TF1.9
- **Bazel version (if compiling from source)**:
N/A
- **CUDA/cuDNN version**:
N/A
- **GPU model and memory**:
N/A
- **Exact command to reproduce**:
gcloud ml-engine jobs submit training object_detection_`date +%m_%d_%Y_%H_%M_%S` \
    --runtime-version 1.9 \
    --job-dir=gs://${MODEL_DIR} \
    --packages dist/object_detection-0.1.tar.gz,slim/dist/slim-0.1.tar.gz,/tmp/pycocotools/pycocotools-2.0.tar.gz \
    --module-name object_detection.model_main \
    --region us-central1 \
    --config ${PATH_TO_LOCAL_YAML_FILE} \
    -- \
    --model_dir=gs://${MODEL_DIR} \
    --pipeline_config_path=gs://${PIPELINE_CONFIG_PATH}

### Describe the problem
Training on GCP and using the model for inference works perfectly. 
However, I'd like to visualise metrics, such as mAP, AR, classification loss and localization loss, on TensorBoard.

### Source code / logs
In "model_main.py", changing eval_on_train_data=True does not add the training data on TensorBoard
```
train_spec, eval_specs = 
model_lib.create_train_and_eval_specs(
            train_input_fn,
            eval_input_fns,
            eval_on_train_input_fn,
            predict_input_fn,
            train_steps,
            eval_on_train_data=True)
```

Furthermore, when looking at the snippet below I understand by providing a checkpoint_dir that it is possible to execute `model_lib.continuous_eval(estimator, FLAGS.checkpoint_dir, input_fn,                               train_steps, name)`
```
if FLAGS.checkpoint_dir:
    if FLAGS.eval_training_data:
      name = 'training_data'
      input_fn = eval_on_train_input_fn
    else:
      name = 'validation_data'
      # The first eval input will be evaluated.
      input_fn = eval_input_fns[0]
    if FLAGS.run_once:
      estimator.evaluate(input_fn,
                         num_eval_steps=None,
                         checkpoint_path=tf.train.latest_checkpoint(
                             FLAGS.checkpoint_dir))
    else:
      model_lib.continuous_eval(estimator, FLAGS.checkpoint_dir, input_fn,
                                train_steps, name)
  else:
    train_spec, eval_specs = model_lib.create_train_and_eval_specs(
        train_input_fn,
        eval_input_fns,
        eval_on_train_input_fn,
        predict_input_fn,
        train_steps,
        eval_on_train_data=False)
```
But the model_dir contains the checkpoints, so I am confused.
So, the question is: Can we evaluate on the training data with as goal to visualize and compare the metrics on Tensorboard?

**Comment by swirlingsand** [comment]
May be and opportunity to improve clarity on _eval_ concepts in object detection api.
It seems like an eval spec is "forced" in the current setup.
Is there some rationale for this that I am missing?

Trying to set eval to use train data results in
`raise ValueError('Image with id {} already added.'.format(image_id))`

**Comment by smivv** [comment]
@swirlingsand shall we maybe add **train_only** flag?
From my point of view is strange that we have possibility to eval in another binary, but no possibility to train without evaluation.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
